apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/McsClaimTransformerV2.java

```diff
@@ -62,78 +84,188 @@ private static Claim transformClaim(PreAdjMcsClaim claimGroup) {
     Claim claim = new Claim();
 
     claim.setId("m-" + claimGroup.getIdrClmHdIcn());
-    claim.setMeta(new Meta().setLastUpdated(Date.from(claimGroup.getLastUpdated())));
-    claim.setContained(getContainedProvider(claimGroup));
+    claim.setContained(List.of(getContainedPatient(claimGroup), getContainedProvider(claimGroup)));
+    claim.setExtension(getExtension(claimGroup));
     claim.setIdentifier(getIdentifier(claimGroup));
-    claim.setStatus(Claim.ClaimStatus.ACTIVE);
+    if (claimGroup.getIdrStatusCode() != null) {
+      claim.setStatus(
+          CANCELED_STATUS_CODES.contains(claimGroup.getIdrStatusCode().toLowerCase())
+              ? Claim.ClaimStatus.CANCELLED
+              : Claim.ClaimStatus.ACTIVE);
+    }
     claim.setType(getType());
+    claim.setBillablePeriod(getBillablePeriod(claimGroup));
     claim.setUse(Claim.Use.CLAIM);
-    claim.setCreated(new Date());
     claim.setPriority(getPriority());
     claim.setTotal(getTotal(claimGroup));
     claim.setProvider(new Reference("#provider-org"));
-    claim.setPatient(getPatient(claimGroup));
+    claim.setPatient(new Reference("#patient"));
     claim.setDiagnosis(getDiagnosis(claimGroup));
-    claim.setProcedure(getProcedure(claimGroup));
+    claim.setItem(getItems(claimGroup));
+
+    FhirContext ctx = FhirContext.forR4();
+    IParser parser = ctx.newJsonParser();
+
+    String claimContent = parser.encodeResourceToString(claim);
+
+    String resourceHash = DigestUtils.sha1Hex(claimContent);
+
+    claim.setCreated(new Date());
+    claim.setMeta(
+        new Meta()
+            .setLastUpdated(Date.from(claimGroup.getLastUpdated()))
+            .setVersionId("f-" + claimGroup.getIdrClmHdIcn() + "-" + resourceHash));
```

Comment:
Also this version Id formatter. Seems some things that could be common-ized?

Reply:
I'm assuming you don't mean the concatenation of three strings, which would be overly trivial to try to make a utility for.

The hashing is specific to the `Claim`, which there are only 2 places that it's being used, not sure where I would put it, it initially seemed almost wasteful to me to create another class for something that is only used twice and is not particularly complex, but I'm open to suggestions.  I'm a big fan of abstracting common logic, but I've also seen it used too eagerly, resulting in fairly bloated code.

Reply:
If it's only two places I wouldn't bother (I typically abide by "don't abstract unless its used 3+ times") just figured it might be a wider pattern since I saw it a couple times in this PR and don't know the other transformer code well yet.